lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -15,6 +15,29 @@ defmodule Mix.Tasks.Compile.Elixir do
   Note it is important to recompile a file's dependencies as
   there are often compile time dependencies between them.
 
+  ## `__mix_recompile__?/0`
+
+  A module may export a `__mix_recompile__?/0` function that can cause the module to be recompiled
```

Comment:
Nitpick: please wrap the lines shorter, as this is larger than the sections before and after.